src/dfcx_scrapi/tools/fulfillments_extraction.py

```diff
@@ -0,0 +1,387 @@
+"""Utility file for working with fulfillments from dialogflow cx."""
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import google.cloud.dialogflowcx_v3beta1.types as types
+import pandas as pd
+
+
+from dfcx_scrapi.core import (
+    scrapi_base,
+    intents,
+    flows,
+    pages,
+    transition_route_groups,
+)
+
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+class Fulfillments(scrapi_base.ScrapiBase):
+    """Class that supports fulfillment functions in DFCX."""
+
+    def __init__(
+        self,
+        creds_path: str = None,
+        creds_dict: dict = None,
+        creds=None,
+        scope=False,
+        agent_id: str = None,
+    ):
+        super().__init__(
+            creds_path=creds_path,
+            creds_dict=creds_dict,
+            creds=creds,
+            scope=scope,
+        )
+
+        logging.info("create dfcx creds %s", creds_path)
+        self.intents = intents.Intents(creds_path, creds_dict)
+        self.flows = flows.Flows(creds_path, creds_dict)
+        self.pages = pages.Pages(creds_path, creds_dict)
+        self.route_groups = transition_route_groups.TransitionRouteGroups(
+            creds_path, creds_dict
+        )
+        self.creds_path = creds_path
+
+        if agent_id:
+            self.agent_id = agent_id
+            self.intents_map = self.intents.get_intents_map(agent_id)
+
+    @staticmethod
+    def combine_dataframes(main: pd.DataFrame, component: pd.DataFrame):
```

Comment:
extraneous

Reply:
@Greenford if you find the opportunity to consolidate code into common methods while reviewing, please do so.